### PR TITLE
Feature/7.x/upgrade helper

### PR DESCRIPTION
**Pull request by Mpdreamz**
This adds a `NEST.7xUpgradeHelper` package. Intended to be released with the `7.0.0` GA release.

We can remove it after the `GA` released during the 7.0 timeline, this is just intended as a one off release.

Referencing this nuget package will reintroduce the methods that have been moved on `IElasticClient` albeit marked `Obsolete` with a message indicating what to rewrite the method invocation too.

### CODE REVIEW

**Comment by russcam** [comment]
Do we want a more generalized name for the package e.g. `NEST.UpgradeAssistant`? Thinking about whether we may want to provide a similar package for future major releases.

**Comment by Mpdreamz** [comment]
Decided to rename to `Nest.7xUpgradeAssistant`

File: build/NEST.7xUpgradeHelper.nuspec [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+		<license type="file">license.txt</license>
+		<projectUrl>https://github.com/elastic/elasticsearch-net</projectUrl>
+		<iconUrl>https://raw.githubusercontent.com/elastic/elasticsearch-net/master/build/nuget-icon.png</iconUrl>
+		<description>NEST 7.0 introduces namespaces on the client, this temporarily reintroduces removed methods on the client with instructions on how to upgrade</description>
+		<summary>NEST 7.0 introduces namespaces on the client, this temporarily reintroduces removed methods on the client with instructions on how to upgrade</summary>
+		<releaseNotes>https://github.com/elastic/elasticsearch-net/releases/tag/$version$</releaseNotes>
+		<tags>elasticsearch,elastic,search,lucene,nest</tags>
+		<dependencies>
+			<group targetFramework=".NETStandard2.0">
+				<dependency id="NEST" version="$version$" />
```
Comment (Mpdreamz) [inline build/NEST.7xUpgradeHelper.nuspec]: Double check this is not fixed

Comment (Mpdreamz) [reply build/NEST.7xUpgradeHelper.nuspec]: Constrain to less then 8.0

[the remark this answers is not in the document]
Comment (Mpdreamz) [reply build/NEST.7xUpgradeHelper.nuspec]: Thank you for catching that! :+1:

**Comment by Mpdreamz** [comment]
No integration test failures on netcoreapp but failed to start on net461 which is fixed #3887

**Pull request merged**